### ISSUE TITLE
get_base_url fails for non-ascii URLs in Python 3

### ISSUE THREAD

**Issue by kmike** [scrapy/scrapy#1783]
E.g. for this page

> http://naftir.com/index.php/%D9%85%D8%AE%D8%A7%D8%B2%D9%86-%D8%B0%D8%AE%DB%8C%D8%B1%D9%87

scrapy.utils.response.get_base_url raises the following exception in Pyhton 3:

```
File "/Users/kmike/svn/scrapy/scrapy/utils/response.py", line 30, in get_base_url
    response.encoding)
  File "/Users/kmike/envs/dl/lib/python3.5/site-packages/w3lib-1.13.0-py3.5.egg/w3lib/html.py", line 287, in get_base_url
    baseurl = moves.urllib.parse.urljoin(baseurl, m.group(1).encode(encoding))
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/urllib/parse.py", line 415, in urljoin
    base, url, _coerce_result = _coerce_args(base, url)
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/urllib/parse.py", line 114, in _coerce_args
    return _decode_args(args) + (_encode_result,)
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/urllib/parse.py", line 98, in _decode_args
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/urllib/parse.py", line 98, in <genexpr>
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd9 in position 28: ordinal not in range(128)
```

**Comment by lopuhin** [comment]
I tried to fix the tests @redapple  added here https://github.com/lopuhin/w3lib/commit/70281a5b58d594fc634713502caaed8a0ae461fe, I'll check a few things and submit a PR if all is ok.

**Comment by redapple** [comment]
https://github.com/scrapy/w3lib/pull/45 fixes this issue for me (with scrapy11rc3):

```
$ scrapy shell "http://naftir.com/index.php/%D9%85%D8%AE%D8%A7%D8%B2%D9%86-%D8%B0%D8%AE%DB%8C%D8%B1%D9%87"
2016-03-23 10:53:11 [scrapy] INFO: Scrapy 1.1.0rc3 started (bot: scrapybot)
(...)
2016-03-23 10:53:12 [scrapy] INFO: Enabled item pipelines:
[]
2016-03-23 10:53:12 [scrapy] INFO: Spider opened
2016-03-23 10:53:13 [scrapy] DEBUG: Crawled (200) <GET http://naftir.com/index.php/%D9%85%D8%AE%D8%A7%D8%B2%D9%86-%D8%B0%D8%AE%DB%8C%D8%B1%D9%87> (referer: None)
(...)
>>> from scrapy.utils.response import get_base_url
>>> get_base_url(response)
'http://naftir.com/index.php/%D9%85%D8%AE%D8%A7%D8%B2%D9%86-%D8%B0%D8%AE%DB%8C%D8%B1%D9%87'
>>> 
```

**Comment by redapple** [comment]
Fixed with latest w3lib v1.14.2

```
$ scrapy version -v
Scrapy    : 1.1.0rc3
lxml      : 3.5.0.0
libxml2   : 2.9.2
Twisted   : 15.5.0
Python    : 2.7.10 (default, Oct 14 2015, 16:09:02) - [GCC 5.2.1 20151010]
pyOpenSSL : 0.15.1 (OpenSSL 1.0.2d 9 Jul 2015)
Platform  : Linux-4.2.0-35-generic-x86_64-with-Ubuntu-15.10-wily

$ pip freeze | grep w3lib
w3lib==1.14.2

$ scrapy shell "http://naftir.com/index.php/%D9%85%D8%AE%D8%A7%D8%B2%D9%86-%D8%B0%D8%AE%DB%8C%D8%B1%D9%87"
2016-04-12 16:38:00 [scrapy] INFO: Scrapy 1.1.0rc3 started (bot: scrapybot)
2016-04-12 16:38:00 [scrapy] INFO: Overridden settings: {'LOGSTATS_INTERVAL': 0, 'DUPEFILTER_CLASS': 'scrapy.dupefilters.BaseDupeFilter'}
(...)
2016-04-12 16:38:02 [scrapy] INFO: Spider opened
2016-04-12 16:38:04 [scrapy] DEBUG: Crawled (200) <GET http://naftir.com/index.php/%D9%85%D8%AE%D8%A7%D8%B2%D9%86-%D8%B0%D8%AE%DB%8C%D8%B1%D9%87> (referer: None)
(...)
In [1]: from scrapy.utils.response import get_base_url

In [2]: get_base_url(response)
Out[2]: 'http://naftir.com/index.php/%D9%85%D8%AE%D8%A7%D8%B2%D9%86-%D8%B0%D8%AE%DB%8C%D8%B1%D9%87'
```
